aws/resource_aws_budgets_budget.go

```diff
@@ -170,9 +204,80 @@ func resourceAwsBudgetsBudgetCreate(d *schema.ResourceData, meta interface{}) er
 	}
 
 	d.SetId(fmt.Sprintf("%s:%s", accountID, *budget.BudgetName))
+
+	notificationsRaw := d.Get("notification").(*schema.Set).List()
+	notifications, subscribers := expandBudgetNotificationsUnmarshal(notificationsRaw)
+
+	err = resourceAwsBudgetsBudgetNotificationsCreate(notifications, subscribers, *budget.BudgetName, accountID, meta)
+
+	if err != nil {
+		return fmt.Errorf("create budget notification failed: %v", err)
+	}
+
 	return resourceAwsBudgetsBudgetRead(d, meta)
 }
 
+func resourceAwsBudgetsBudgetNotificationsCreate(notifications []*budgets.Notification, subscribers [][]*budgets.Subscriber, budgetName string, accountID string, meta interface{}) error {
+	client := meta.(*AWSClient).budgetconn
+
+	for i, notification := range notifications {
+		subscribers := subscribers[i]
+		if len(subscribers) == 0 {
+			return fmt.Errorf("Notification must have at least one subscriber!")
+		}
+		_, err := client.CreateNotification(&budgets.CreateNotificationInput{
+			BudgetName:   aws.String(budgetName),
+			AccountId:    aws.String(accountID),
+			Notification: notification,
+			Subscribers:  subscribers,
+		})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func expandBudgetNotificationsUnmarshal(notificationsRaw []interface{}) ([]*budgets.Notification, [][]*budgets.Subscriber) {
+
+	notifications := make([]*budgets.Notification, len(notificationsRaw))
+	subscribersForNotifications := make([][]*budgets.Subscriber, len(notificationsRaw))
+	for i, notificationRaw := range notificationsRaw {
+		notificationRaw := notificationRaw.(map[string]interface{})
+		comparisonOperator := notificationRaw["comparison_operator"].(string)
+		threshold := notificationRaw["threshold"].(float64)
+		thresholdType := notificationRaw["threshold_type"].(string)
+		notificationType := notificationRaw["notification_type"].(string)
+
+		notifications[i] = &budgets.Notification{
+			ComparisonOperator: aws.String(comparisonOperator),
+			Threshold:          aws.Float64(threshold),
+			ThresholdType:      aws.String(thresholdType),
+			NotificationType:   aws.String(notificationType),
+		}
+
+		emailSubscribers := expandSubscribers(notificationRaw["subscriber_email_addresses"], budgets.SubscriptionTypeEmail)
+		snsSubscribers := expandSubscribers(notificationRaw["subscriber_sns_topic_arns"], budgets.SubscriptionTypeSns)
+
+		subscribersForNotifications[i] = append(emailSubscribers, snsSubscribers...)
+	}
+	return notifications, subscribersForNotifications
+}
+
+func expandSubscribers(rawList interface{}, subscriptionType string) []*budgets.Subscriber {
```

Comment:
Nit: Since this codebase uses a very flat Go package structure, we should include additional information in the function signature so it doesn't collide with other potential usage, e.g.

```suggestion
func expandBudgetSubscribers(rawList interface{}, subscriptionType string) []*budgets.Subscriber {
```